### PR TITLE
fixes for 0.7

### PR DESCRIPTION
**Pull request by rened**
Passes the tests for 0.6 and 0.7 without warnings.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -159,25 +160,26 @@ module NullableTestEnum
 end
 
 if VERSION < v"0.7.0-DEV.2797"
-    # showcompact(io::IO, x::Nullable)
+    showc(io, x) = show(IOContext(io, :compact => true), x)
```
Comment (nalimilan) [inline test/runtests.jl]: This only applies to old versions: it should probably continue to use `showcompact`, right?

File: src/nullable.jl [hunk 2]
```diff
@@ -427,7 +427,7 @@ _nullable_eltype(f, A, As...) =
 @inline function Base.broadcast(f, ::Base.Broadcast.Style{Nullable}, ::Nothing, ::Nothing, a...)
     nonnull = all(hasvalue, a)
     S = _nullable_eltype(f, a...)
-    if Base._isleaftype(S) && null_safe_op(f, Base.Broadcast.maptoTuple(_unsafe_get_eltype,
+    if Base.isconcretetype(S) && null_safe_op(f, Base.Broadcast.maptoTuple(_unsafe_get_eltype,
                                                                         a...).types...)
```
Comment (nalimilan) [inline src/nullable.jl]: Keep arguments aligned.

**Review by nalimilan — COMMENTED** [review]
Thanks. Ideally I think the `showcompact` support should be adapted to use the new `typeinfo` property, but if people really care they can always implement it later.

File: test/runtests.jl (hunk 1 above)
Comment (rened) [reply test/runtests.jl]: True! Fixed.

**Pull request merged**